packages/components/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,263 @@
+  headerA11y?: string
+  /** Optional indeterminate boolean. Overrides checked styling */
+  indeterminate?: boolean
+  /** Optional accessibility override for label  */
+  labelA11y?: string
+  /** Optional required boolean that denotes a label as (*Required) */
+  required?: boolean
+  /** Optional tile styling */
+  tile?: boolean
+}
+
+export const Checkbox: FC<CheckboxProps> = ({
+  checked,
+  label,
+  labelA11y,
+  description,
+  descriptionA11y,
+  error,
+  errorA11y,
+  header,
+  headerA11y,
+  hint,
+  hintA11y,
+  indeterminate,
+  onPress,
+  required,
+  tile,
+}) => {
+  const { t } = useTranslation()
+  const theme = useTheme()
+
+  const baseTextStyle = {
+    fontFamily: 'SourceSansPro-Regular',
+    color: theme.vadsColorForegroundDefault,
+  }
+
+  /**
+   * Container styling
+   */
+
+  let containerStyle: ViewStyle = {
+    width: '100%',
+  }
+
+  if (error) {
+    containerStyle = {
+      ...containerStyle,
+      borderLeftWidth: spacing.vadsSpace2xs,
+      borderColor: theme.vadsColorFormsBorderError,
+      paddingLeft: spacing.vadsSpaceLg,
+    }
+  }
+
+  /**
+   * Pressable styling
+   */
+
+  const pressableBaseStyle: StyleProp<ViewStyle> = {
+    width: '100%',
+    flexDirection: 'row',
+    alignItems: 'flex-start',
+  }
+
+  const tileStyle: ViewStyle = {
+    ...pressableBaseStyle,
+    borderWidth: 2,
+    borderRadius: spacing.vadsSpace2xs,
+    paddingTop: spacing.vadsSpaceSm,
+    paddingBottom: spacing.vadsSpaceSm,
+    paddingLeft: spacing.vadsSpaceSm,
+    paddingRight: spacing.vadsSpaceMd,
+    borderColor: checked
+      ? theme.vadsColorFormsBorderActive
+      : theme.vadsColorFormsBorderSubtle,
+    backgroundColor: checked
+      ? theme.vadsColorFormsSurfaceActive
+      : theme.vadsColorSurfaceDefault,
+  }
+
+  /**
+   * Icon
+   */
+
+  const iconViewStyle: ViewStyle = {
+    alignItems: 'center',
+    justifyContent: 'center',
+    paddingTop: 3,
+    paddingBottom: 3,
+  }
+
+  const _icon = (
+    <View style={iconViewStyle}>
+      <Icon
+        name={
+          indeterminate
+            ? 'IndeterminateCheckBox'
+            : checked
+              ? 'CheckBox'
+              : 'CheckBoxOutlineBlank'
+        }
+        fill={
+          checked
+            ? theme.vadsColorFormsForegroundActive
+            : theme.vadsColorFormsBorderDefault
+        }
+      />
+    </View>
+  )
+
+  /**
+   * Label
+   */
+
+  const labelProps = {
+    'aria-label': labelA11y || label,
+    style: {
+      ...baseTextStyle,
+      fontFamily: error ? 'SourceSansPro-Bold' : 'SourceSansPro-Regular',
+      fontSize: 20,
+      lineHeight: 30,
+    },
+  }
+
+  const requiredStyle = {
+    color: theme.vadsColorForegroundError,
+  }
+
+  const _label = (
+    <Text {...labelProps}>
+      {label}
+      {required && <Text style={requiredStyle}>{` (*${t('required')})`}</Text>}
+    </Text>
+  )
+
+  /**
+   * Header
+   */
+
+  const headerProps = {
+    'aria-label': headerA11y || header,
```

Comment:
Since it's applied directly to the text, not sure if we need the `||` or if it being undefined will automatically defer to the text; could potentially simplify here and other spots.